modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionPayloads.scala

```diff
@@ -0,0 +1,44 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import cats.data.NonEmptyList
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.api.construction.ConstructionMetadata.TransactionMetadata
+import org.tessellation.rosetta.domain.operation.Operation
+import org.tessellation.rosetta.domain.{NetworkIdentifier, RosettaPublicKey, SigningPayload}
+import org.tessellation.security.hex.Hex
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{customizableDecoder, customizableEncoder}
+import derevo.derive
+
+case object ConstructionPayloads {
+
+  @derive(customizableDecoder)
+  case class Request(
+    networkIdentifier: NetworkIdentifier,
+    operations: NonEmptyList[Operation],
+    metadata: Option[TransactionMetadata],
```

Comment:
@sdavidp Why did you decide to make it optional?
according to rosetta specification:

> Payloads is called with an array of operations and the response from `/construction/metadata`.

while `/construction/metadata` specifies `metadata` as required property, so there is no way (IMO) to pass here `None` 🤔

Reply:
yes, because there are implementations/networks that don't require additional metadata, so you would not pass metadata. but we require metadata (for example: last reference) so it should be a required field.

Reply:
rosetta API specification is written to cover all possibilities, while we are writing an implementation for specific network, not generic one. the API should be crafted for our network then. by doing that, you would automatically not allow to accept a request that is invalid (because for example has no metadata, so you won't be able to construct a transaction).

Reply:
I agree with Marcin. The rosetta specs are very generic which means that the concrete implementation should have more strict types tailored for our case. Of course it should still be a subset of rosetta generic types.

Reply:
ok. thank you for the clarification.